dash/development/_py_components_generation.py

```diff
@@ -65,11 +65,14 @@ def __init__(self, {default_argtext}):
         _explicit_args = kwargs.pop('_explicit_args')
         _locals = locals()
         _locals.update(kwargs)  # For wildcard attrs and excess named props
-        args = {{k: _locals[k] for k in _explicit_args if k != 'children'}}
+        args = {args}
         for k in {required_props}:
             if k not in args:
                 raise TypeError(
                     'Required argument `' + k + '` was not specified.')
+        if 'children' in {required_props} and 'children' not in _explicit_args:
+            raise TypeError(
+                    'Required argument children was not specified.')
```

Comment:
Can we test `'children' in required_props` at generation time and omit this block entirely if not?

(I'd love it if we could also omit the `for k in {required_props}` block above when there are no required props - that's not new but bonus points if you're up for implementing that 😁)

Reply:
We applied conditional generating args in https://github.com/plotly/dash/pull/2218/commits/bf71a22e5a3c54afa9c9c5b44960b29f4a2277b0 and https://github.com/plotly/dash/pull/2218/commits/316e53049b85bb8803fbd6e9f151b94d5b16d486

And also we add test case gene004 in https://github.com/plotly/dash/pull/2218/commits/b802c5dd5435d54dca95f80fe1e105c234bd948a